src/Magnum/GL/Test/CMakeLists.txt

```
@@ -162,10 +170,10 @@ if(BUILD_GL_TESTS)
     endif()
 
     if(NOT MAGNUM_TARGET_GLES2)
-        corrade_add_test(GLBufferImageGLTest BufferImageGLTest.cpp LIBRARIES MagnumOpenGLTesterTestLib)
+        corrade_add_test(GLBufferImageGLTest BufferImageGLTest.cpp LIBRARIES MagnumDebugTools MagnumOpenGLTesterTestLib)
```

To fix the Mac CI failure, can you try swapping the order?

```suggestion
        corrade_add_test(GLBufferImageGLTest BufferImageGLTest.cpp LIBRARIES MagnumOpenGLTesterTestLib MagnumDebugTools)
```

Same below. The `*TestLib` has the GL library built with graceful assertions, while the regular `DebugTools` links to a regular `MagnumGL`. Meaning, there's two duplicate candidates for the (global) GL symbols and the linker picks the first it finds (from `DebugTools`), but then the code in `MagnumOpenGLTesterTestLib` relies on the second duplicate, causing an assertion due to a different copy of the global used in different parts of the code.

I hope that's enough to make this work, if not then I'd probably need to introduce `MagnumDebugToolsTestLib` that relies on `MagnumGLTestLib` instead of `MagnumGL`.

---

Ah, thanks, I understand the issue now. Sadly changing the order doesn't seem to make a difference 😢 I assume CMake sorts the link order anyway